go/arrow/schema.go

```diff
@@ -222,6 +222,18 @@ func (sc *Schema) Equal(o *Schema) bool {
 	return true
 }
 
+func (s *Schema) AddField(i int, field Field) (*Schema, error) {
```

Comment:
Like with the table, would `InsertField` be a better name? Personally I see `Add` and immediately think of adding to the end rather than inserting.